formatter/README.md

```diff
@@ -0,0 +1,148 @@
+# AUTOMATED CODE FORMATTING
+
+## BACKGROUND
+
+Code Formatting is considered "best practice" as it provides code quality assurance for a given project. Project code is consistently formatted in the same manner throughout the project tree and all development is directed to the same formatting standard. Linting and formatting source code files prior to committing to a CVS also removes extraneous whitespace characters. Differences in formatting style and extraneous whitespaces typically creep into a project over time where multiple developers are working on a project and each developer is using their preferred editing application. Problem is compounded when developers are using multiple platforms (i.e. Unix, Linux, Windows, et al)
+
+All necessary configuration settings for automated code formatting as it applies to **DB Browser for SQLite** are stored in this folder.
+
+## DEPENDENT APPLICATIONS
+
+Application leverages Python 3 to execute other applications to perform automated code formatting. These dependent applications are available across may platforms Ensure these applications are installed prior to attempting to follow the **OPERATION** instructions below.
+
+### Python 3
+
++ available for most platforms
++ Unix and Linux distributions may already have it installed.
++ see <https://www.python.org/downloads/> to install for your platform
+
+### toml
+
++ `Tom's Obvious Markup Language` is used for format configuration storage.
++ Python does not have a _standard_ module for toml parsing.
++ install module from pypi using `pip install toml`
++ toml project - <https://github.com/toml-lang/toml>
++ wikipedia - <https://en.wikipedia.org/wiki/TOML>
++ pypi - <https://pypi.org/project/toml/>
+
+### astyle
+
++ small, automatic formatter for C, C++, Objective-C, and Java languages
++ available for most platforms
++ project and documentation - <http://astyle.sourceforge.net/astyle.html>
++ Sourceforge - <https://sourceforge.net/projects/astyle/>
++ configuration - `astylerc`
+
+### xmllint
+
++ basic XML file linting and formatting
++ available for most platforms
++ apart of the libxml2 library
++ project - <http://www.xmlsoft.org/>
++ configuration - environment variables and flags defined in settings.
+
+### cmake-format
+
++ Python module containing QA tools for CMake files
++ install via `pip install cmake_format`
++ project - <https://github.com/cheshirekow/cmake_format>
++ pypi - <https://pypi.org/project/cmake-format/>
++ documentation - <https://cmake-format.readthedocs.io/en/latest/>
++ configuration - `cmake-format.json`
+
+## OPERATION
+
+For all platforms, invoke Python on the module `format_runner.py`:
+
+```bash
+[cd|DIR] [path to DB4S project root]
+python3 formatter/format_runner.py
```

Comment:
Will that process all files in the directory? Isn't that going too far? The file to correct should be passed as command line argument, or as much, the tool should only process changed files according to git.

Reply:
The invocation above is to start process for all tools to do work. What they work on is limited to the files|patterns identified in the `format_runner.toml` file for a particular format tool. Have a look at the `[[tools]]` entry for `xmllint`. It only does work on two files. More can be added. Case in point, `astyle`. If the `recursive` switch isn't used it only works on the files in a directory with pattern given.